### DISCUSSION TITLE
source-controller CrashLoopBackOff issue

### DISCUSSION THREAD

**Discussion by shridhar-at-droit** [fluxcd/flux2#1512] [Q&A]
I have noticed below error in `source-controller` logs 

`E0611 23:15:29.839798       9 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.4/tools/cache/reflector.go:167: Failed to watch *v1beta1.GitRepository: failed to list *v1beta1.GitRepository: request to convert CR from an invalid group/version: source.toolkit.fluxcd.io/v1alpha1  

{"level":"info","ts":"2021-06-11T23:15:29.913Z","logger":"controller.bucket","msg":"Starting Controller","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"Bucket"}

E0611 23:15:30.902961       9 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.4/tools/cache/reflector.go:167: Failed to watch *v1beta1.GitRepository: failed to list *v1beta1.GitRepository: request to convert CR from an invalid group/version: source.toolkit.fluxcd.io/v1alpha1

{"level":"error","ts":"2021-06-11T23:17:29.814Z","logger":"controller.helmrepository","msg":"Could not wait for Cache to sync","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"HelmRepository","error":"failed to wait for helmrepository caches to sync: cache did not sync"}

{"level":"error","ts":"2021-06-11T23:17:29.814Z","logger":"controller.bucket","msg":"Could not wait for Cache to sync","reconciler group":"source.toolkit.fluxcd.io","reconciler kind":"Bucket","error":"failed to wait for bucket caches to sync: cache did not sync"}

{"level":"error","ts":"2021-06-11T23:17:29.814Z","msg":"error received after stop sequence was engaged","error":"failed to wait for bucket caches to sync: cache did not sync"}

{"level":"error","ts":"2021-06-11T23:17:29.814Z","msg":"error received after stop sequence was engaged","error":"Timeout: failed waiting for *v1beta1.GitRepository Informer to sync"}

{"level":"error","ts":"2021-06-11T23:17:29.814Z","msg":"error received after stop sequence was engaged","error":"Timeout: failed waiting for *v1beta1.GitRepository Informer to sync"}

{"level":"error","ts":"2021-06-11T23:17:29.814Z","logger":"setup","msg":"problem running manager","error":"failed to wait for helmrepository caches to sync: cache did not sync"}`

**Comment by stefanprodan** [comment]
Support for `source.toolkit.fluxcd.io/v1alpha1` has been dropped months ago, you need to uninstall and bootstrap your cluster with the latest flux CLI.

**Reply by shridhar-at-droit** [reply]
Hello @stefanprodan ,
I did exactly the same, I uninstall Flux v1 and installed fresh Flux v2, but not sure why it still referring to `source.toolkit.fluxcd.io/v1alpha1`.

i also checked `kubectl api-versions` and `kubectl api-resources` but did not find any reference to `source.toolkit.fluxcd.io/v1alpha1`.

**Reply by hiddeco** [reply]
Any chance you are making use of k3s? We have had a report in the past (which I can not find at the moment) about the API backing storage getting corrupt and the v1alpha1 API deceleration still being present there.

**Reply by shridhar-at-droit** [reply]
I am using EKS, and flux v1 was working fine..

**Reply by hiddeco** [reply]
Please see the steps in https://github.com/fluxcd/kustomize-controller/issues/316

**Reply by shridhar-at-droit** [reply]
@hiddeco 
I stuck to this step `Delete all the resources that are now listed under kubectl get kustomizations -A`
it is in hang state, and I tried to patch `finalizers`
```
kubectl patch kustomization/axxxx-env -n gitops-system  \
  --type=merge \
  --patch '{"metadata":{"finalizers":[]}}'
```
but I got error response The `Kustomization "xxxx-env" is invalid: status.snapshot.checksum: Required value`
I tried  to do edit manually but it  did not accepted the changes.

**Reply by hiddeco** [reply]
Your original errors are about source resources, you should thus follow the steps but modify the values so that they apply to that (source) API, and not the Kustomization one.

**Reply by shridhar-at-droit** [reply]
@hiddeco 
just to clarify that i have issue with. both resources, `source` and `kustomization` but kustomization pod was not restarting like source pod.
Also noticed that after applying `kustomization` v0.0.13 , kustomization list reappeared even the namespace gitops-system is deleted.

```
NAMESPACE       NAME                READY       STATUS                                                                AGE
gitops-system   xxx-dev              True       Applied revision: master/7499be4a6cf4ae76ac4a79041ca1e6f69b4afe54     262d
gitops-system   xxx                  False      artifact acquisition failed                                           235d
gitops-system   yy-dev               True       Applied revision: master/88b64432805cfbe743f2bf4ebb5ba67eae210bda    267d
gitops-system   zz-dev               True       Applied revision: master/a0a72c8894b2f25d98ce84f7d1306349a5abf5c4    262d
```
 I followed the same steps for source and reinstalled flux v2 but still i got the the same issue

**kustomization error:**
`E0614 09:54:01.023026       6 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.4/tools/cache/reflector.go:167: Failed to watch *v1beta1.Kustomization: failed to list *v1beta1.Kustomization: request to convert CR from an invalid group/version: kustomize.toolkit.fluxcd.io/v1alpha1`

**source error:**
`E0614 09:53:45.009991       6 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.4/tools/cache/reflector.go:167: Failed to watch *v1beta1.GitRepository: failed to list *v1beta1.GitRepository: request to convert CR from an invalid group/version: source.toolkit.fluxcd.io/v1alpha1`

In flux V2  `kubectl get kustomization -A`  response is as following
`Error from server: request to convert CR from an invalid group/version: kustomize.toolkit.fluxcd.io/v1alpha1`

**Reply by shridhar-at-droit** [reply]
I managed to cleanup old kustomization list by removing checksum references   in v0.0.13  crd (as suggested in the link provided)  and resinstall flux v2 but i still see those error (in both)

**Reply by hiddeco** [reply]
Are you sure you do not have a reference to a `v1alpha1` resource anywhere in the repository you are synchronizing from?

**Reply by shridhar-at-droit** [reply]
`grep -r apiVersion . | grep -i toolkit | awk '{ print $2 }' | sort | uniq -c   `
   5 kustomize.toolkit.fluxcd.io/v1beta1
   5 source.toolkit.fluxcd.io/v1beta1

**kubectl get kustomization -A**
```
NAMESPACE     NAME          READY   STATUS   AGE
flux-system   flux-system                    2m31s
```

**Reply by shridhar-at-droit** [reply]
There is change in the error in kustomization

**Now it is showing same error as Source** 

`E0614 11:01:48.347267       6 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.4/tools/cache/reflector.go:167: Failed to watch *v1beta1.GitRepository: failed to list *v1beta1.GitRepository: request to convert CR from an invalid group/version: source.toolkit.fluxcd.io/v1alpha1`

**and Source error is** 

`E0614 11:02:52.319416       6 reflector.go:138] pkg/mod/k8s.io/client-go@v0.20.4/tools/cache/reflector.go:167: Failed to watch *v1beta1.GitRepository: failed to list *v1beta1.GitRepository: request to convert CR from an invalid group/version: source.toolkit.fluxcd.io/v1alpha1`


I believe above steps fixed the kustomization related error but Source error still persist.
